### PR TITLE
Help prevent hard fail on nil undefined legacy networks.

### PR DESCRIPTION
**Pull request by chris1984**
PR to work to prevent this with a hard fail:

https://bugzilla.redhat.com/show_bug.cgi?id=1612921

### CODE REVIEW

File: lib/fog/vsphere/requests/compute/list_vm_interfaces.rb [hunk 1]
```diff
@@ -62,10 +62,17 @@ def get_raw_interface(vm_id, options = {})
         private
 
         def raw_to_hash(nic)
+          if nic.backing.respond_to?('network')
```
Comment (timogoebel) [inline lib/fog/vsphere/requests/compute/list_vm_interfaces.rb]: While we're at it, can you please change `'network'` to a symbol (`:network`)? Same for port below.

**Comment by chris1984** [comment]
@timogoebel thanks for the review :) Do you want me to change it to make rubopcop happy or want me to add a line to disable the cop for this if block and renable it after? The latter would be easier

**Comment by chris1984** [comment]
@timogoebel I went with the latter, updated if you want to merge.

**Comment by timogoebel** [comment]
Thanks, @chris1984.

**Pull request merged**